src/code/engine_library/runMeanModelWorkflow.m

```diff
@@ -0,0 +1,234 @@
+function runMeanModelWorkflow(WSettings,TaskList,MeanModelSet,VPC,Datafiles)
+% RUNMEANMODELWORKFLOW master routine for population workflow
+%
+% runMeanModelWorkflow(TaskList,WSettings,PopRunSet,varargin)
+% 
+% Inputs:
+%       WSettings (structure)    definition of properties used in all
+%                   workflow functions see GETDEFAULTWORKFLOWSETTINGS
+%       TaskList (structure)    list of task which should be executed see GENERATEWORKFLOWINPUTFORMEANMODELSIMULATION
+%       RUNMEANMODELWORKFLOW (structure)   list of population simulations see GENERATEWORKFLOWINPUTFORMEANMODELSIMULATION
+%       VPC  (structure) contains information which plots should be
+%               generated  see GETDEFAULTVPCPOPULATIONSETTINGS
+
+% Open Systems Pharmacology Suite;  http://open-systems-pharmacology.org
+
+
+% try
+    %% check optional inputs
+    if ~exist('Datafiles','var')
+        Datafiles = {};
+    end
+
+    %% initialize workflow
+    [WSettings] = initializeWorkflow('Mean Model Simulation',WSettings);
+    
+    %% check if necessary inputs for task are available
+    writeToLog(sprintf('Start input checks'),WSettings.logfile,true,false);
+    
+    successInputCheck = true;
+
+    % simulation
+    for iSet = 1:length(MeanModelSet)
+        successInputCheck = checkMeanModelInput(WSettings,MeanModelSet(iSet)) & successInputCheck;
+    end
+
+
+    % data TP file
+    % check if data read in is necessary
+    doReadDataTPfile = TaskList.doVPC & ~isempty(Datafiles);
+    
+    % check file name of data read in
+    if doReadDataTPfile
+        successInputCheck = checkInputDatafiles(WSettings,Datafiles) & successInputCheck;
+    end
+
+    % PK Parameter
+    if TaskList.doVPC
+        if isempty(VPC)
+            writeToLog(sprintf('ERROR: VPC definition is missing!'),WSettings.logfile,true,false);
+            successInputCheck = false;
+        end
+    end
+    
+    
+    % stop if not all inputs are available
+    if ~successInputCheck
+        return
+    end
+
+    
+    % process xmls, check for consistency  and create temporary
+    % directory with infos like application protocols, and factor for
+    % unitconversion, and timeprofiles of desired outputs 
+
+    for iSet = 1:length(MeanModelSet)
+        [success,VPC] = prepareMeanModelSimulation(WSettings,MeanModelSet(iSet),VPC);
+        successInputCheck = success & successInputCheck;
+    end
```

Comment:
here ALL models are processed even if _prepareMeanModelSimulation_ fails on the very first model.
on purpose?